text/0097-log-data-model.md

```diff
@@ -0,0 +1,988 @@
+    <td>body</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>any</td>
+    <td>Structured data.</td>
+    <td>attributes</td>
+  </tr>
+</table>
+
+### Apache HTTP Server access log
+
+<table>
+  <tr>
+    <td>Field</td>
+    <td>Type</td>
+    <td>Description</td>
+    <td>Maps to Unified Model Field</td>
+  </tr>
+  <tr>
+    <td>%t</td>
+    <td>timestamp</td>
+    <td>Time when an event occurred measured by the origin clock.</td>
+    <td>timestamp</td>
+  </tr>
+  <tr>
+    <td>%a</td>
+    <td>string</td>
+    <td>Client IP</td>
+    <td>attributes["net.peer.ip"]</td>
+  </tr>
+  <tr>
+    <td>%A</td>
+    <td>string</td>
+    <td>Server IP</td>
+    <td>attributes["net.host.ip"]</td>
+  </tr>
+  <tr>
+    <td>%h</td>
+    <td>string</td>
+    <td>Remote hostname. </td>
+    <td>attributes["net.peer.name"]</td>
+  </tr>
+  <tr>
+    <td>%m</td>
+    <td>string</td>
+    <td>The request method.</td>
+    <td>attributes["http.method"]</td>
+  </tr>
+  <tr>
+    <td>%v,%p,%U,%q</td>
+    <td>string</td>
+    <td>Multiple fields that can be composed into URL.</td>
+    <td>attributes["http.url"]</td>
+  </tr>
+  <tr>
+    <td>%>s</td>
+    <td>string</td>
+    <td>Response status.</td>
+    <td>attributes["http.status_code"]</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>any</td>
+    <td>Structured data.</td>
+    <td>attributes, key=TBD</td>
+  </tr>
+</table>
+
+### CloudTrail Log Event
+
+<table>
+  <tr>
+    <td>Field</td>
+    <td>Type</td>
+    <td>Description</td>
+    <td>Maps to Unified Model Field</td>
+  </tr>
+  <tr>
+    <td>eventTime</td>
+    <td>string</td>
+    <td>The date and time the request was made, in coordinated universal time (UTC).</td>
+    <td>timestamp</td>
+  </tr>
+  <tr>
+    <td>eventSource</td>
+    <td>string</td>
+    <td>The service that the request was made to. This name is typically a short form of the service name without spaces plus .amazonaws.com.</td>
+    <td>resource["service.name"]?</td>
+  </tr>
+  <tr>
+    <td>awsRegion</td>
+    <td>string</td>
+    <td>The AWS region that the request was made to, such as us-east-2.</td>
+    <td>resource["cloud.region"]</td>
+  </tr>
+  <tr>
+    <td>sourceIPAddress</td>
+    <td>string</td>
+    <td>The IP address that the request was made from.</td>
+    <td>resource["net.peer.ip"] or resource["net.host.ip"]? TBD</td>
+  </tr>
+  <tr>
+    <td>errorCode</td>
+    <td>string</td>
+    <td>The AWS service error if the request returns an error.</td>
+    <td>short_description</td>
+  </tr>
+  <tr>
+    <td>errorMessage</td>
+    <td>string</td>
+    <td>If the request returns an error, the description of the error.</td>
+    <td>body</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>*</td>
+    <td></td>
+    <td>attributes["cloudtrail.*"]</td>
+  </tr>
+</table>
+
+## Appendix B: `SeverityNumber` example mappings
+
+|Syslog       |WinEvtLog  |Log4j |Zap   |java.util.logging|SeverityNumber|
```

Comment:
Could you add these mappings from Google Cloud Logging's logs' severity levels to the universal data model's severity levels? @otoolep for visibiltiy.

Defualt -> DEBUG
Debug -> DEBUG2
Info -> INFO
Notice -> INFO2
Warning -> WARN
Error -> Error
Critical -> Error2
Alert -> Error3
Emergency -> Fatal

Reply:
Thanks @jsandler18

Specifically this is how we would map this proposal to the GCP Cloud Logging Data model as per here: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry

Reply:
Is Default->DEBUG desirable? The docs says that Default means "The log entry has no assigned severity level."

Is the idea that Default should be treated as the least severe log record? In that case perhaps TRACE is a better choice for mapping?

Alternatively if Default means unknown should it be mapped to a more serious level, e.g. INFO?

Reply:
@tigrannajaryan this leads to a question: should we have an "unknown" possibility where we let the backend decide what to do with that? This is for sources that don't have a severity level or they support this option.

Reply:
It doesn't look like all the mappings above have severity mappings here, so I think we'll actually punt on this for now.

Reply:
@bogdandrutu Yes. Severity is an optional field. We can either define an explicit "unknown" value or state that if Severity is missing then it is "unknown". I was intending to do the later but I think either way works.

Reply:
@jsandler18 was thinking "Default" as not defined.

Reply:
I am keeping the notion that "unknown" is the same as "missing" for now. This means that "Default" of GCP Cloud Logging Data will result in missing Severity fields during mapping.

Should we find that this is incorrect we can add "Unknown" as a value that is distinct from "missing" later.